pkg/lib/cockpit-components-terminal.jsx

```diff
@@ -237,10 +232,12 @@ export class Terminal extends React.Component {
         }
     }
 
-    disconnectChannel() {
-        if (this.props.channel) {
-            this.props.channel.removeEventListener('message', this.onChannelMessage);
-            this.props.channel.removeEventListener('close', this.onChannelClose);
+    disconnectChannel(channel) {
+        if (channel === undefined)
+            channel = this.props.channel;
```

Comment:
Is it ok to assign this here? It's an anti-practice to change props variable and this one is coming from prevProps.

Reply:
Ah, haven't thought about it, I just used it as local variable. I surely can create one local variable which will be either channel and when not defined than would use current channel from props.

Reply:
That was a wrong alarm from my side.
Since you change the parameter itself , that won't affect the item that was fed into the parameter.
If you would change the internals of the parameter (aka object values), that will propagate back up, causing problems, since that would affect the props.